components/theia/packages/gitpod-extension/src/browser/gitpod-user-storage-provider.ts

```diff
@@ -93,6 +94,49 @@ export class GitpodUserStorageProvider implements FileSystemProviderWithFileRead
         }]);
     }
 
+    // To avoid having badly written extensions executing too much requests and putting pressure on other components we
+    // cache writes and reads here.
+    protected readonly contentCache = new Map<string, string>();
+    protected updatedSinceLastWrite = new Set<string>();
+    protected async cachedWrite(resource: GitpodServer.UpdateUserStorageResourceOptions): Promise<void> {
+        this.contentCache.set(resource.uri, resource.content);
+        this.updatedSinceLastWrite.add(resource.uri);
```

Comment:
would it make sense to track clusters of write operations per uri and log them? so we might get some insights on the issue.

Reply:
I was thinking about https://developer.mozilla.org/en-US/docs/Web/API/Console/table we can on every submit of writes to the server. an accumulated count would interesting, too.

Reply:
@AlexTugarev Does this work for you? https://github.com/gitpod-io/gitpod/pull/2476/commits/57a21c9f6d96cf905c02f9c4b2056d30b378d207